### PR TITLE
test infrastructure and payment request tests

### PR DESCRIPTION
**Pull request by petersalomonsen**
# Test infrastructure setup

- Switch to "BOS workspace" as test gateway
- RPC mocking
- Transaction submit mocking
- "Don't ask again" cache config

# test: create payment request

https://github.com/user-attachments/assets/a5300a97-1369-430a-840e-d85c9ce45d4a

# test: create manual payment request

https://github.com/user-attachments/assets/7d770aa9-e6f5-4927-b1b6-fb4896fdf900

# test: approve payment request ( "don't ask again" enabled )

https://github.com/user-attachments/assets/a9ec8cd9-0503-4eb0-92f3-715a0a4e31b8

### CODE REVIEW

**Comment by petersalomonsen** [comment]
@Megha-Dev-19 See the "approve payment request" test video, where "Don't ask again" is enabled, but the loading spinner on the approve button does not stop. Is this missing for real, or does it only occur in the test?

**Comment by petersalomonsen** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
### ⏸️ This repository is on pause
@petersalomonsen please reach out to the maintainer to unpause this repo with `@race-of-sloths unpause` command

File: playwright-tests/tests/payments/create-payment-request.spec.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+} from "../../util/transaction";
+import { mockRpcRequest } from "../../util/rpcmock";
+import { setDontAskAgainCacheValues } from "../../util/cache";
+
+test.describe("admin connected", function () {
+  test.use({
+    storageState: "playwright-tests/storage-states/wallet-connected-admin.json",
+  });
+  test("create payment request", async ({ page }) => {
+    await page.goto("/treasury-devdao.near/widget/app?page=payments");
```
Comment (Megha-Dev-19) [inline playwright-tests/tests/payments/create-payment-request.spec.js:14]: should we also add test for manual request?

**Review by Megha-Dev-19 — APPROVED** [review]
LGTM

**Comment by Megha-Dev-19** [comment]
> @Megha-Dev-19 See the "approve payment request" test video, where "Don't ask again" is enabled, but the loading spinner on the approve button does not stop. Is this missing for real, or does it only occur in the test?

Yeah it's specific to local gateway, I have shared the exact details on chat with you

File: playwright-tests/tests/payments/create-payment-request.spec.js (hunk 1 above)
Comment (petersalomonsen) [reply playwright-tests/tests/payments/create-payment-request.spec.js:14]: yes, I will add that too

**Comment by petersalomonsen** [comment]
@Megha-Dev-19 See the updated test video for approving payment requests with "don't ask again" enabled. This shows that the table is empty after the transaction is completed, and also that the approved payment request can be found under the history tab. Can you confirm if this is the correct and expected behaviour, and that we can keep the test like this?

**Comment by Megha-Dev-19** [comment]
@petersalomonsen yes that’s the expected behaviour 👍

File: playwright-tests/tests/payments/create-payment-request.spec.js (hunk 1 above)
Comment (petersalomonsen) [reply playwright-tests/tests/payments/create-payment-request.spec.js:59]: @Megha-Dev-19 See the test ( you can find the video in the description ) for creating manual payment request

Comment (Megha-Dev-19) [reply playwright-tests/tests/payments/create-payment-request.spec.js:59]: @petersalomonsen it looks great

**Pull request merged**